RA-14-SeekBar

DenisGubernatorov opened the pull request:
1) Реализация seekBar для количества порций

DenisGubernatorov committed:
RA-14-SeekBar

1) Реализация seekBar для количества порций

aGroupoid commented on app/src/main/res/layout/recipe_fragment.xml:
```diff
@@ -48,34 +48,65 @@
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/margin_full_16_dp"
+                android:max="4"
+                android:progress="0">
```
Давай установим min, progress = 1 параметры, как ты сделал для max. Тогда не придется прибавлять эту непонятную магическую единичку в листенере

DenisGubernatorov replied:
Получается что теперь допустимое состояние 0 порций. 
![image](https://github.com/user-attachments/assets/dcca4adc-f741-4132-8785-9a0842208a12)
Как-то не очень логично на мой взгляд. Как минимум пишется количество ингридиентов на 1 порцию

aGroupoid commented on app/src/main/java/com/example/recipesapp/IngredientsAdapter.kt:
```diff
@@ -7,17 +7,31 @@ import android.view.ViewGroup
+            return if (value.rem(1.0) == 0.0) {
+                DecimalFormat("#,###").format(value.toInt())
+            } else {
+                DecimalFormat("#,###.#").format(value)
```
Вообще для чисел с точкой лучше использовать BigDecimal. Он позволяет точнее сравнивать и производить вычисления. Плюс есть встроенные методы для форматирования строк. Статья про BigDecimal: https://javarush.com/groups/posts/2274-kak-ispoljhzovatjh-bigdecimal-v-java

val totalQuantity = BigDecimal(ingredientCurrentCount) * BigDecimal(quantity)

val displayQuantity = totalQuantity
.setScale(1,RoundingMode.HALF_UP)
.stripTrailingZeros()
.toPlainString()

aGroupoid commented on app/src/main/java/com/example/recipesapp/RecipeFragment.kt:
```diff
@@ -57,19 +58,38 @@ class RecipeFragment : Fragment() {
 
+        binding.SeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            val offset = 1
+            override fun onProgressChanged(p0: SeekBar?, p1: Int, p2: Boolean) {
```
Те параметры, которые используешь - нужно назвать внятными именами, p1 - непонятно что значит

DenisGubernatorov committed:
RA-14-SeekBar

1) Получение количества ингредиентов в зависимости от количества переведено на использование BigDecimal
2) Отказ от использования смещения для правильного расчета количества порций
3) Переименование аргументов в onProgressChanged

aGroupoid commented:
По заданию всё верно, молодец! Поправь в следующем ПР, я в комменте написал, что надо также выставить min = 1 в xml, тогда мы ограничем порции от 1
<img width="237" alt="Снимок экрана 2024-12-27 в 12 21 18" src="https://github.com/user-attachments/assets/a8405a6d-03dd-42bb-b933-03dde3354610" />
